packages/nimble-components/src/table/specs/table-column-specs/table-column-mapping.md

```diff
@@ -0,0 +1,359 @@
+
+Text in a grouping header will be ellipsized and gain a tooltip if there is not enough room to display it all.
+
+### Sorting
+
+Sorting will be based on the record value. For boolean and number values, a basic sort (just using basic comparison/equality operators) is the clear choice. For string values, it is less clear. In the case where the strings are enum values, they are likely to be non-localized, English strings. In that case, even if there is a semantically meaningful sort order (i.e. "NOT_STARTED" < "RUNNING" < "DONE"), we can't sort that way. Settling for a basic sort in this case seems just as reasonable as anything else. If the strings are localized, it's more likely the client would use a text column instead. If we did want to support a different sorting method for strings than booleans/numbers, it would be awkward to implement. We would have update the column's sort method when the data changes, and the column would have to determine the data type for its field, which might require looking at multiple records (e.g. if the first didn't define that field). For these reasons, we will just use a basic sort for all supported data types (string, boolean and number).
+
+For icons, if multiple values map the to same icon, it is possible that sorting will result in the instances of a certain icon not being all together in one span of rows. The user may think they should be all together, but this is a corner case that we can't/shouldn't do anything about.
+
+### Sizing
+
+`nimble-table-column-icon` will support only a fixed width. We will introduce a new mixin for fixed-width support that exposes a `pixel-width` property. The default value will be the minimum supported by the table, which is still significantly larger than the width of an icon.
+
+`nimble-table-column-mapping` will support fixed or fractional widths. If `pixel-width` is set, the column will have a fixed width, otherwise it defaults to a fractional width of 1. The client may configure `fractional-width` and/or `min-pixel-width`.
+
+### Angular integration
+
+Angular directives will be created for the column components and the mapping components. No component has form association, so a `ControlValueAccessor` will not be created.
+
+### Blazor integration
+
+Blazor wrappers will be created for the components. Mapping components will be generic in the type of the key:
+
+```HTML
+<NimbleTableColumnMapping Field="NumberData" DataType="number">
+    <NimbleMappingText TKey=int Key="1" Label="foo"></NimbleMappingText>
+    <NimbleMappingText TKey=int Key="2" Label="bar"></NimbleMappingText>
```

Comment:
Is this generic API useful now? If it were always a "string" wouldn't the underlying `nimble-table-column-mapping` still do the right thing because its `datatype` is set to `number`? As is, it seems like we've opened the door for a mismatch of types between the `NimbleTableColumnMapping` and its contained mapping elements.

Reply:
I've updated the spec to indicate we'll be using `CascadingTypeParameter`.